packages/central-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -168,32 +168,56 @@ export async function importSurveyResponses(req, res) {
       for (let columnIndex = minSurveyResponseIndex; columnIndex <= maxColumnIndex; columnIndex++) {
         const columnHeader = getColumnHeader(sheet, columnIndex);
         const importMode = getImportMode(columnHeader);
+        const entityCode = getInfoForColumn(sheet, columnIndex, 'Entity Code');
+        const entityName = getInfoForColumn(sheet, columnIndex, 'Entity Name');
+        const entity = await models.entity.findOne({ code: entityCode });
+
+        if (entityCode && entityName) {
+          if (entity) {
+            if (entity?.name !== entityName) {
+              throw new Error(
+                `Entity code and name don't match: ${entity?.name} and ${entityName}`,
+              );
+            }
+          } else {
+            throw new Error(`Entity code does match any existing entity: ${entityCode}`);
+          }
+        }
 
         if (IMPORT_BEHAVIOURS[importMode].shouldGenerateIds) {
-          surveyResponseIds[columnIndex] = generateId();
+          surveyResponses[columnIndex] = { surveyResponseId: generateId(), entityId: entity?.id };
           isGeneratedIdByColumnIndex[columnIndex] = true;
         } else if (IMPORT_BEHAVIOURS[importMode].shouldUpdateExistingResponses) {
           const { surveyResponseId } = await getExistingResponseData(columnIndex);
 
           if (surveyResponseId) {
-            surveyResponseIds[columnIndex] = surveyResponseId;
+            surveyResponses[columnIndex] = {
+              surveyResponseId: surveyResponseId,
+              entityId: entity?.id,
+            };
           } else {
             // A matching existing response was not found, generate a new id
-            surveyResponseIds[columnIndex] = generateId();
+            surveyResponses[columnIndex] = {
+              surveyResponseId: generateId(),
+              entityId: entity?.id,
+            };
             isGeneratedIdByColumnIndex[columnIndex] = true;
           }
         } else {
-          surveyResponseIds[columnIndex] = columnHeader;
+          surveyResponses[columnIndex] = {
+            surveyResponseId: columnHeader,
+            entityId: entity?.id,
+          };
```

Comment:
I think there's probably a way we can reduce repetition here. Perhaps by doing something like `let surveyResponseIdValue` and then in each case setting it to the correct value, and then after all the `if/else` statements, setting:
```
surveyResponses[columnIndex] = {
   surveyresponseId: surveyResponseIdValue,
   entityId: entity?.id
}

```